common/djangoapps/third_party_auth/api/permissions.py

```diff
@@ -0,0 +1,29 @@
+"""
+Third party auth API related permissions
+"""
+from rest_framework import permissions
+
+from third_party_auth.models import ApiPermissions
+
+
+class ThirdPartyAuthApiPermission(permissions.BasePermission):
+    """
+    Allow someone to access the view if they have valid OAuth client credential.
+    """
+    def has_permission(self, request, view):
+        """
+        Check if the OAuth client associated with auth token in current request has permission to access
+        the information for provider
+        """
+        provider_id = request.parser_context['kwargs'].get('provider_id', None)
```

Comment:
Nit: Consider passing the `provider_id` in the `__init__` method instead of assuming it's a certain field in the request object.